spec/models/company_spec.rb

```diff
@@ -1,5 +1,78 @@
 require 'rails_helper'
 
-RSpec.describe Company, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+describe Company do
+  context 'Receving datas' do
+    context '#token' do
+      it 'should generate a token on create' do
+        company = build(:company)
+
+        company.save
+
+        expect(company.token).to_not be_blank
+      end
+
+      it 'token must be unique' do
+        company = build(:company)
+        another_company = create(:company)
+        allow(SecureRandom).to receive(:alphanumeric).and_return(another_company.token, 'ABC123')
+        company.save
+        expect(company.token).not_to eq(another_company.token)
+      end
+    end
+
+    context '#cnpj' do
+      it 'canot be blank' do
```

Comment:
```suggestion
      it 'cannot be blank' do
```

Aqui tem um pequeno typo hehe